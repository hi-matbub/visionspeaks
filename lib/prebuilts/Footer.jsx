import React from 'react';
import PropTypes from 'prop-types';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import SlimWrapper from '../wrappers/SlimWrapper';
import PrimaryModal from './PrimaryModal';
import Wrap from '../components/Wrap';
import FAQ from '../static/FAQ.mdx';
import About from '../static/About.mdx';
import TermsAndConditions from '../static/TermsAndConditions.mdx';
import ShippingAndReturns from '../static/ShippingAndReturns.mdx';
import PrivacyPolicy from '../static/PrivacyPolicy.mdx';
import Pill from '../components/Pill';

const date = new Date();
const year = date.getFullYear();

const Footer = ({ data }) => {
  const { store } = data;
  return (
    <footer>
      <Container className="my-5">
        <SlimWrapper>
          <h2>More Info</h2>
          <Wrap>
            <div className="d-flex flex-column">
              <PrimaryModal title="About">
                <About />
              </PrimaryModal>
              <PrimaryModal title="FAQ">
                <FAQ />
              </PrimaryModal>
              <a
                href={`mailto:${store.contact}`}
                target="_blank"
                rel="noreferrer"
              >
                <Pill title="Contact" />
              </a>
            </div>
            <div className="d-flex flex-column">
              <PrimaryModal title="Terms and Conditions">
                <TermsAndConditions />
              </PrimaryModal>
              <PrimaryModal title="Privacy Policy">
                <PrivacyPolicy />
              </PrimaryModal>
              <PrimaryModal title="Shipping and Returns">
                <ShippingAndReturns />
              </PrimaryModal>
            </div>
          </Wrap>
        </SlimWrapper>

        <Row>
          <Col className="d-flex justify-content-center">
            <p className="m-5 text-center">
              Copyright <br /> {year}
            </p>
          </Col>
        </Row>
      </Container>
    </footer>
  );
};

Footer.propTypes = {
  data: PropTypes.shape({
    store: PropTypes.shape({
      contact: PropTypes.shape({}).isRequired,
    }).isRequired,
  }).isRequired,
};

export default Footer;
