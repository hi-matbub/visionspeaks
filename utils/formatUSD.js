/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {
  // for inconsistent types
  const string = JSON.stringify(parseFloat(number));
  const { length } = string || number;
  const str = `${string.substring(0, length - 2)}.${string.substring(
    length - 2
  )}`;
  return parseFloat(str).toFixed(2);
};

export { formatUSD };
