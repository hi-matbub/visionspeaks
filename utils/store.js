exports.store = {
  name: 'Dog-Gone Goods',
  logo: '/logo.svg',
  hero: '/hero.svg',
  url: '',
  description: 'A dog treat shop',
  about: 'Everyone loves a happy dog!',
  contact: 'youremail@here.com',
  social: {
    facebook: '',
    twitter: '',
    instagram: '',
  },
};
