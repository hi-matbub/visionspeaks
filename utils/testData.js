// used for testing purposes

exports.testData = {
  store: {
    name: 'Dog-Gone Goods',
    logo: '/logo.svg',
    hero: '/hero.svg',
    url: '',
    description: 'A dog treat shop',
    about: 'Everyone loves a happy dog!',
    contact: 'youremail@here.com',
    social: { facebook: '', twitter: '', instagram: '' },
  },
  products: [
    {
      id: 'prod_IRbpyL6fgse3n',
      object: 'product',
      active: true,
      attributes: [],
      created: 1606151601,
      description: 'demo product description',
      images: ['img.jpg'],
      livemode: false,
      metadata: {
        price: 287,
      },
      name: 'Large Demo Product',
      statement_descriptor: null,
      type: 'service',
      unit_label: null,
      updated: 1606153202,
    },
    {
      id: 'prod_IRbouvicOUom16',
      object: 'product',
      active: true,
      attributes: [],
      created: 1606151551,
      description: 'demo product description',
      images: ['img.jpg'],
      livemode: false,
      metadata: {
        price: 287,
      },
      name: 'Demo Product',
      statement_descriptor: null,
      type: 'service',
      unit_label: null,
      updated: 1606153221,
    },
  ],
};
