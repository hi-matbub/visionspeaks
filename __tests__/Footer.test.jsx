import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Footer from '../lib/prebuilts/Footer';
import { testData } from '../utils/testData';

it('Footer component renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Footer data={testData}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Footer component renders unchanged', () => {
  const tree = renderer.create(<Footer data={testData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
