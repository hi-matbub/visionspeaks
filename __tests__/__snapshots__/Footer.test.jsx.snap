// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component renders unchanged 1`] = `
<footer>
  <div
    className="my-5 container"
  >
    <div
      className="align-items-center justify-content-center row"
    >
      <div
        className="col-lg-8 col-md-9 col-sm-12"
      >
        <h2>
          More Info
        </h2>
        <div
          className="d-flex flex-wrap"
        >
          <div
            className="d-flex flex-column"
          >
            <button
              className="btn btn-outline-dark mt-2 pl-1 text-left"
              id="About"
              onClick={[Function]}
              style={
                Object {
                  "border": "0",
                  "padding": "0",
                  "width": "200px",
                }
              }
              type="button"
            >
              About
            </button>
            <button
              className="btn btn-outline-dark mt-2 pl-1 text-left"
              id="FAQ"
              onClick={[Function]}
              style={
                Object {
                  "border": "0",
                  "padding": "0",
                  "width": "200px",
                }
              }
              type="button"
            >
              FAQ
            </button>
            <a
              href="mailto:youremail@here.com"
              rel="noreferrer"
              target="_blank"
            >
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="Contact"
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                Contact
              </button>
            </a>
          </div>
          <div
            className="d-flex flex-column"
          >
            <button
              className="btn btn-outline-dark mt-2 pl-1 text-left"
              id="Terms and Conditions"
              onClick={[Function]}
              style={
                Object {
                  "border": "0",
                  "padding": "0",
                  "width": "200px",
                }
              }
              type="button"
            >
              Terms and Conditions
            </button>
            <button
              className="btn btn-outline-dark mt-2 pl-1 text-left"
              id="Privacy Policy"
              onClick={[Function]}
              style={
                Object {
                  "border": "0",
                  "padding": "0",
                  "width": "200px",
                }
              }
              type="button"
            >
              Privacy Policy
            </button>
            <button
              className="btn btn-outline-dark mt-2 pl-1 text-left"
              id="Shipping and Returns"
              onClick={[Function]}
              style={
                Object {
                  "border": "0",
                  "padding": "0",
                  "width": "200px",
                }
              }
              type="button"
            >
              Shipping and Returns
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="d-flex justify-content-center col"
      >
        <p
          className="m-5 text-center"
        >
          Copyright 
          <br />
           
          2020
        </p>
      </div>
    </div>
  </div>
</footer>
`;
